spark-plugins/src/main/java/io/cdap/plugin/spark/DataCacher.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © 2016-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.spark;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.batch.SparkCompute;
+import io.cdap.cdap.etl.api.batch.SparkExecutionPluginContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.storage.StorageLevel;
+
+/**
+ * SparkCompute that caches a RDD
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name("DataCacher")
+@Description("Spark Data Cacher caches any incoming records and outputs them unchanged.")
+public class DataCacher extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+  private final DataCacherConfig config;
+
+  public DataCacher(DataCacherConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(
+    SparkExecutionPluginContext sparkExecutionPluginContext, JavaRDD<StructuredRecord> javaRDD)
+    throws Exception {
+
+    StorageLevel storageLevel = config.getStorageLevel();
+    if (storageLevel == StorageLevel.NONE()) {
```

Comment:
This should be a validation error that should be checked at deployment time.

Reply:
We cant check at deployment time because the spark classes are not available during deployment/validation. The plugin uses a dropdown to specify the storage level so the only way the user can pass an incorrect value is using a macro which we cant check during deployment anyway.

Reply:
Maybe check that the string value is in one of the allowed value.

Reply:
Sure, done.